src/signals.c

```diff
@@ -298,43 +298,69 @@ static RETSIGTYPE sig_abort(int signo) {
 }
 
 static RETSIGTYPE sig_terminate(int signo) {
+  const char *signame = "(unsupported)";
+  int log_signal = TRUE, log_stacktrace = TRUE;
+
   /* Capture the signal number for later display purposes. */
   term_signo = signo;
 
-  if (signo == SIGSEGV ||
-      signo == SIGXCPU
+  /* Some terminating signals get more special treatment than others. */
+
+  switch (signo) {
+    case SIGSEGV:
+      recvd_signal_flags |= RECEIVED_SIG_SEGV;
+      signame = "SIGSEGV";
+      break;
+
+    case SIGXCPU:
+      recvd_signal_flags |= RECEIVED_SIG_XCPU;
+      signame = "SIGXCPU";
+      break;
+
+    case SIGTERM:
+      /* Since SIGTERM is more common, we do not want to log as much for it. */
+      log_signal = log_stacktrace = FALSE;
+      recvd_signal_flags |= RECEIVED_SIG_TERMINATE;
+      signame = "SIGTERM";
+      break;
+
 #ifdef SIGBUS
-      || signo == SIGBUS) {
-#else
-     ) {
+    case SIGBUS:
+      recvd_signal_flags |= RECEIVED_SIG_TERMINATE;
+      signame = "SIGBUS";
+      break;
 #endif /* SIGBUS */
 
-    if (signo == SIGXCPU) {
-      recvd_signal_flags |= RECEIVED_SIG_XCPU;
+    case SIGILL:
+      recvd_signal_flags |= RECEIVED_SIG_TERMINATE;
+      signame = "SIGILL";
+      break;
 
-    } else {
-      recvd_signal_flags |= RECEIVED_SIG_SEGV;
-    }
+    case SIGINT:
+      recvd_signal_flags |= RECEIVED_SIG_TERMINATE;
+      signame = "SIGINT";
+      break;
 
+    default:
+      break;
```

Comment:
why it would be bad here to set `recvd_signal_flags |= RECEIVED_SIG_TERMINATE;` ? because I see, that some other signals like `SIGTRAP` is leaving children hanging up too